src/tracker/long_tracker.jl

```diff
@@ -0,0 +1,161 @@
+    match_total = MatchedPairs(props2)
+    while true # there are no more floes to match in props1
+
+        # This routine mutates both props1 and props2.
+        # Get preliminary matches for floe r in props1 from floes in props2
+
+        # Container for props of matched floe pairs and their similarity ratios. Matches will be updated and added to match_total
+        matched_pairs = MatchedPairs(props2)
+        for r in 1:nrow(props1) # TODO: consider using eachrow(props1) to iterate over rows
+            # 1. Collect preliminary matches for floe r in matching_floes
+            matching_floes = makeemptydffrom(props2)
+
+            for s in 1:nrow(props2) # TODO: consider using eachrow(props2) to iterate over rows
+                Δt = get_dt(props1, r, props2, s)
+                @debug "Considering floe 2:$s for floe 1:$r"
+                ratios, conditions, dist = compute_ratios_conditions(
+                    (props1, r), (props2, s), Δt, condition_thresholds
+                )
+
+                if callmatchcorr(conditions)
+                    @debug "Getting mismatch and correlation for floe 1:$r and floe 2:$s"
+                    (area_mismatch, corr) = matchcorr(
+                        props1.mask[r], props2.mask[s], Δt; mc_thresholds.comp...
+                    )
+
+                    if isfloegoodmatch(
+                        conditions, mc_thresholds.goodness, area_mismatch, corr
+                    )
+                        @debug "** Found a good match for floe 1:$r => 2:$s"
+                        appendrows!(
+                            matching_floes,
+                            props2[s, :],
+                            (ratios..., area_mismatch, corr),
+                            s,
+                            dist,
+                        )
+                        @debug "Matching floes" matching_floes
+                    end
+                end
+            end # of s for loop
+
+            # 2. Find the best match for floe r
+            @debug "Finding best match for floe 1:$r"
+            best_match_idx = getidxmostminimumeverything(matching_floes.ratios)
+            @debug "Best match index for floe 1:$r: $best_match_idx"
+            if isnotnan(best_match_idx)
+                bestmatchdata = getbestmatchdata(
+                    best_match_idx, r, props1, matching_floes
+                ) # might be copying data unnecessarily
+                addmatch!(matched_pairs, bestmatchdata)
+                @debug "Matched pairs" matched_pairs
+            end
+        end # of for r = 1:nrow(props1)
+
+        # exit while loop if there are no more floes to match
+        @debug "Matched pairs" matched_pairs
+        isempty(matched_pairs) && break
+
+        #= Resolve collisions:
+        Are there floes in day k+1 paired with more than one
+        floe in day k? If so, keep the best matching pair and remove all others. =#
+        resolvecollisions!(matched_pairs)
```

Comment:
I'm not sure about the name `collisions`. I don't think you're talking about physical collisions between floes, I think you're talking about multiple matches. Is that correct?

Reply:
Yes, by `collision` I am referring to two or more things being associated to the same object, not things physically clashing with each other. I believe `collision` is a standard term that captures this situation in computer science lingo. See for example https://en.wikipedia.org/wiki/Hash_collision.

Reply:
I'm aware of the meaning of the term in computing. But this is a program about physical objects which can collide too, and their locations in space are the whole point of the code, so the term "collide" may be interpreted in two different ways, and the first one which came to my mind was the physical one.